lib/ae_mdw/aex9.ex

```diff
@@ -40,13 +39,14 @@ defmodule AeMdw.Aex9 do
 
   @type amounts :: map()
 
-  @spec fetch_balances(State.t(), pubkey(), boolean()) :: {:ok, amounts()} | {:error, Error.t()}
-  def fetch_balances(state, contract_pk, top?) do
+  @spec fetch_balances(State.t(), State.t(), pubkey(), boolean()) ::
+          {:ok, amounts()} | {:error, Error.t()}
+  def fetch_balances(state, async_state, contract_pk, top?) do
```

Comment:
This change, and all that follows, where `async_state` is explicitly passed (and stored, in the case of `AsyncStoreServer`) are made so that the state can be reliably mocked and not relied on the global one.